extensions/virtual-threads/runtime/src/main/java/io/quarkus/virtual/threads/VirtualThreadsRecorder.java

```diff
@@ -0,0 +1,166 @@
+package io.quarkus.virtual.threads;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.ShutdownContext;
+import io.quarkus.runtime.annotations.Recorder;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.ContextInternal;
+
+@Recorder
+public class VirtualThreadsRecorder {
+
+    private static final Logger logger = Logger.getLogger("io.quarkus.virtual-threads");
+
+    static VirtualThreadsConfig config = new VirtualThreadsConfig();
+
+    private static volatile Executor current;
+    private static final Object lock = new Object();
+
+    public void setupVirtualThreads(VirtualThreadsConfig c, ShutdownContext shutdownContext, LaunchMode launchMode) {
+        config = c;
+        if (launchMode == LaunchMode.DEVELOPMENT) {
+            shutdownContext.addLastShutdownTask(new Runnable() {
+                @Override
+                public void run() {
+                    Executor executor = current;
+                    if (executor instanceof ExecutorService) {
+                        ((ExecutorService) executor).shutdownNow();
```

Comment:
did you check it does not break the dev mode?

Reply:
Yes, it'll simply interrupt the threads. On reload a new service will be created.